src/main/java/com/clevertap/stormdb/StormDB.java

```diff
@@ -718,6 +801,49 @@ private void iterate(final boolean useLatestWalFile, final boolean readInMemoryB
         }
     }
 
+    private void flushDeletedKeys() throws IOException {
+        // flush the deleteSet keys to file
+        rwLock.writeLock().lock();
+        try {
+            if (deleteKeysOut == null || deletedKeysSet.isEmpty()) {
+                return;
+            }
+
+            ByteBuffer deletedKeysBuffer = ByteBuffer.allocate(4 * deletedKeysSet.size());
```

Comment:
If we have 100k deleted keys, we'll allocate 400k memory. Can we do without this?

Reply:
Yeah I can change this also by setting the keysRead bitset itself in this function

Reply:
This cannot be larger than 80 bytes because I am saving every 20 keys into the file

Reply:
Got it.